### DISCUSSION TITLE
fix: Failed storing data in memcached, error code: 47

### DISCUSSION THREAD

**Discussion by Sosiashvili** [phalcon/phalcon#242] [Q&A]
Hello Phalcon team;

Today I got some error about memcache:

Failed storing data in memcached, error code: 47

some hosting change host requires ...
<pre>
["servers" => [
   [
    "host"   => "/opt/memcached/run/{username}/memcached-1.sock",  **// that before was 127.0.0.1**
    "port"   => 11211,
    "weight" => 1,
  ],
],
</pre>

**good luck...**

example code for native  #php:
<pre>

$mc = new Memcached();
$mc->addServer('/opt/memcached/run/username/memcached-1.sock', 0) or die ("Unable to connect");

echo "Server version:";
print_r($mc->getVersion());

$tmp = new stdClass;
$tmp->str_attr = 'test';
$tmp->int_attr = rand(0,1000);

$mc->set('testkey', $tmp, 10) or die ("Unable to save data in the cache");

$result = $mc->get('testkey');

echo "Data from the cache:";
var_dump($result);

<pre>

**Comment by niden** [comment]
What is the problem? 

Looking at the code, the `Libmemcached` storage adapter uses `addServers()` to add whatever servers you pass to them, and that includes socket based hosts. The only requirement is that you need to have memcached 2.0.0b1 or greater for this to work.
